mbuild/formats/gomc_conf_writer.py

```diff
@@ -0,0 +1,4065 @@
+                      + 'from the selected force field.',
+        ensemble_typ : str, only accepts 'NVT', 'NPT', 'GEMC_NPT', 'GCMC-NVT', 'GCMC'
+        RunSteps : int; must be an integer greater than zero.
+            Sets the total number of simulation steps.
+        Temperature : Temperature of system in Kelvin (K)
+
+        input_variables_dict: dict, default = None
+            These input variables are optional and override the default settings.
+            Changing these variables likely required for more advanced systems.
+
+            The details of the acceptable input variables for the selected
+            ensembles can be found by running this python workbook,
+
+                print_valid_ensemble_input_variables('GCMC', description = True)
+
+            which prints the input_variables with their subsection description
+            for the selected 'GCMC' ensemble (other ensembles can be set as well).
+
+            Example : input_variables_dict = {'Restart' : False, 'PRNG' : 123,
+                                              'ParaTypeCHARMM' : True }
+
+        Returns
+        -------
+        None
+
+        Notes
+        -------
+        The details of the required inputs for the selected
+        ensembles can be found by running this python workbook,
+
+             print_valid_required_input_variables('NVT', description = True)
+
+        which prints the required inputs with their subsection description
+        for the selected 'NVT' ensemble (other ensembles can be set as well).
+
+        The box units imported are in nm.  They need and are to be converted to Ang for GOMC or NAMD
+        """
+
+        # set this to check and see if all the input pass
+        self.all_inputs_pass = True
+
+        # set this to check and see if all the input pass
+        self.all_failed_input_List = []
+
+        # Check if charmm_object is really a Charmm() object
+        if isinstance(charmm_object, mf_charmm.Charmm) == False:
+            self.all_inputs_pass = False
+            print_error_message = 'The variable supplied as a charmm_object ({}}) is not a ' \
+                                  'charmm_object ({}})'.format(type(mf_charmm.Charmm), type(mf_charmm.Charmm))
+            raise ValueError(print_error_message)
+
+        # check ensemble is a correct type
+        print('INFO: ensemble_type = ' + str(ensemble_type))
+        if ensemble_type in ['NPT', 'NVT', 'GCMC', 'GEMC_NVT', 'GEMC_NPT']:
+            self.ensemble_type = ensemble_type
+        else:
+            self.all_inputs_pass = False
+            print_error_message = "The ensemble_type is not a valid entry. "\
+                                  "Only 'NPT', 'NVT', 'GCMC', 'GEMC_NVT', 'GEMC_NPT' are valid entries."
+            raise ValueError(print_error_message)
+
+        self.RunSteps = RunSteps
+        self.Temperature = Temperature
+        if charmm_object.FF_filename != None and isinstance(charmm_object.FF_filename, str) == True:
```

Comment:
Based on [these lines in charmm_writer.py](https://github.com/mosdef-hub/mbuild/pull/831/files#diff-5dd36870d38b79d8983742c98898ebe3906228ff00fd927b648ab8270525ae55R599)

These tests should not be necessary, since the charmm object will have validated this already

Reply:
Yes, in a typical case I agree.  I added them there as an advanced user may specify them in the original Charmm object, but later change them manually in their custom script to redirect the same (charmm.ff_filename = "new_name"), or slightly modified version to a different location.  This would avoid rebuilding the whole system.  Therefore, I would like to leave them in there to catch these cases.